src/pubtools/pulplib/_impl/client/client.py

```diff
@@ -1131,3 +1150,20 @@ def check_repo(repo_on_server):
         out = f_flat_map(out, check_repo)
 
         return f_proxy(out)
+
+    @staticmethod
+    def _check_unit(unit_type, fields, expected_unit, current_unit, repo_id):
```

Comment:
The method and variable naming here seems a bit off to me.  I think `unit` has a specific meaning in Pulp, being a mongo document representing a single piece of content.  I don't think the other types of objects like repositories, distributors, importers are considered "units".  May I suggest something like "resource" might be better?